### DISCUSSION TITLE
After switching and executing multiple workflows through the API, LoRA becomes ineffective. All models are version 1.5.

### DISCUSSION THREAD

**Discussion by prog-ape** [comfyanonymous/ComfyUI#4481] [Q&A]
{
  "8": {
    "inputs": {
      "samples": [
        "131",
        3
      ],
      "vae": [
        "130",
        4
      ]
    },
    "class_type": "VAEDecode",
    "_meta": {
      "title": "VAE Decode"
    }
  },
  "9": {
    "inputs": {
      "filename_prefix": "ComfyUI",
      "images": [
        "8",
        0
      ]
    },
    "class_type": "SaveImage",
    "_meta": {
      "title": "Save Image"
    }
  },
  "130": {
    "inputs": {
      "ckpt_name": "MoyouArtificial.safetensors",
      "vae_name": "Baked VAE",
      "clip_skip": -1,
      "lora_name": "dunhuang.safetensors",
      "lora_model_strength": 0.6,
      "lora_clip_strength": 1,
      "positive": "Epic CG masterpiece, hdr,dtm, full ha,8K, ultra detailed graphic tension, dynamic poses, stunning colors, 3D rendering, surrealism, cinematic lighting effects, realism, 00 renderer, super realistic, full - body photos, super vista, super wide Angle, HD\f\nDunhuang\f1girl\fsolo\fjewelry\flong hair\fhair ornament\fblurry background\fMilky skin, (shiny skin:1.6)\f\npetals\ffalling petals\fnavel\fearrings\fbracelet\fbare shoulders\fstrapless\fred nails \fnail polish\fbandeau\f\nDunhuang style, Dunhuang mural background, influenced by ancient chinese art, daz3d, colorful costumes, Cyan red ribbon, uhd image, nostalgic paintings, tondo\f\nA shot with tension\f(sky glows red,Visual impact,giving the poster a dynamic and visually striking appearance:1.2),Chinese Zen style,impactful picture,\n(dark shot:1), shadows, darkness, rays of light, reflections, mist, smoke, halos, flickering, contrast, ",
      "negative": "EasyNegative, bad-hands-5, text, watermark, low quality, medium quality, blurry, censored, deformed, mutated,",
      "token_normalization": "none",
      "weight_interpretation": "A1111",
      "empty_latent_width": 768,
      "empty_latent_height": 768,
      "batch_size": 1
    },
    "class_type": "Efficient Loader",
    "_meta": {
      "title": "Efficient Loader"
    }
  },
  "131": {
    "inputs": {
      "seed": 1123172560138223,
      "steps": 40,
      "cfg": 7,
      "sampler_name": "dpmpp_2m_sde",
      "scheduler": "karras",
      "denoise": 0.75,
      "preview_method": "none",
      "vae_decode": "true",
      "model": [
        "130",
        0
      ],
      "positive": [
        "130",
        1
      ],
      "negative": [
        "130",
        2
      ],
      "latent_image": [
        "130",
        3
      ]
    },
    "class_type": "KSampler (Efficient)",
    "_meta": {
      "title": "KSampler (Efficient)"
    }
  }
}

Using the latest ComfyUI, the output logs are:
got prompt
Requested to load SD1ClipModel
Loading 1 new model
----------------------------------------
Efficient Loader Models Cache:
Ckpt:
  [1] MoyouArtificial
Lora:
  [1] base_ckpt: MoyouArtificial
      lora(mod,clip): dunhuang(0.6,1.0)
KSampler(Efficient) Warning: No vae input detected, proceeding as if vae_decode was false.

100%|██████████| 40/40 [00:03<00:00, 12.59it/s]
Prompt executed in 6.16 seconds

Does this have anything to do with caching? After each API call to execute a workflow, I added an idle task to clear it. However, after about ten minutes, the LoRA issue reappears.

**Comment by prog-ape** [comment]
Has no one encountered this problem?

**Comment by ltdrdata** [comment]
To be honest, it's difficult to answer because the situation you're describing is not clear.

First, we should clarify whether you clearly understand the workflow execution and cache system.
The basic principles of ComfyUI's execution system are as follows:

1. The prompt of the workflow transmitted to the execution system is passed in its entirety, not just a part, everytime.
2. The execution system runs each node based on the entire prompt.
3. The output generated as a result of node execution is cached.
4. Nodes with already cached outputs are not re-executed; instead, the cached output is utilized.
5. The cache of nodes with changed inputs (including widget values) is cleared. (In cases where inputs are explicitly controlled through IS_CHANGED in the node's definition, the defined change detection logic is used)
6. The effect of cache clearing is propagated transitively.
7. In each prompt execution, nodes are considered the same only if the node ID and type are identical. (In other words, if the node ID changes between two workflows, that node is considered deleted and a new node added, thus treated as a different node.)
